sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ScalaUDF.scala

```diff
@@ -23,6 +23,12 @@ import org.apache.spark.sql.catalyst.expressions.codegen._
 import org.apache.spark.sql.types.DataType
 
 /**
+ * Common base trait for user-defined functions, including ScalaUDF, ScalaUDAF, PythonUDF,
+ * HiveSimpleUDF, HiveGenericUDF, and HiveUDAFFunction.
+ */
+trait UDFType
```

Comment:
maybe UserDefinedExpression?

Reply:
also sort of weird to put this in ScalaUDF.scala. Either put it in Expression.scala, or create its own file ...